src/Server.hs

```diff
@@ -0,0 +1,29 @@
+{-# LANGUAGE DataKinds     #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE RankNTypes    #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Server (api, app) where
```

Comment:
I guess it's better to put it into `Hintman` folder – `Hintman.Server`.

Also, alignment here is a bit strange. We ussually do
```haskell
module Foo
       ( foo
       , bar
       ) where
```